src/boot/boot.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+function injectStylesheet(doc, href) {
+  var link = doc.createElement('link');
+  link.rel = 'stylesheet';
+  link.type = 'text/css';
+  link.href = href;
+  doc.head.appendChild(link);
+}
+
+function injectScript(doc, src) {
+  var script = doc.createElement('script');
+  script.type = 'text/javascript';
+  script.src = src;
+
+  // Set 'async' to false to maintain execution order of scripts.
+  // See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script
+  script.async = false;
+  doc.head.appendChild(script);
+}
+
+function injectAssets(doc, config, assets) {
+  assets.forEach(function (path) {
+    var url = config.assetRoot + 'build/' + config.manifest[path];
+    if (url.match(/\.css/)) {
+      injectStylesheet(doc, url);
+    } else {
+      injectScript(doc, url);
+    }
+  });
+}
+
+/**
+ * Bootstrap the Hypothesis client.
+ *
+ * This triggers loading of the necessary resources for the client
+ */
+function bootHypothesisClient(doc, config) {
+  // Detect presence of Hypothesis in the page
+  var appLinkEl = doc.querySelector('link[type="application/annotator+html"]');
+  if (appLinkEl) {
+    return;
+  }
+
+  // Register the URL of the sidebar app which the Hypothesis client should load.
+  // The <link> tag is also used by browser extensions etc. to detect the
+  // presence of the Hypothesis client on the page.
+  var baseUrl = doc.createElement('link');
+  baseUrl.rel = 'sidebar';
+  baseUrl.href = config.sidebarAppUrl;
```

Comment:
The browser extension still relies on this `<link>` element for the moment. I agree, in future a custom element would probably be a more direct way to indicate the presence of the client on the page.